org.eclipse.jdt.ls.product/scripts/jdtls

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+import glob
+import os
+import platform
+import re
+import semver
+import subprocess
+import sys
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+def get_java_executable():
+	java_executable = 'java'
+
+	if 'JAVA_HOME' in os.environ:
+		java_exec_to_test = os.path.join(os.environ['JAVA_HOME'], 'bin', 'java')
+		if os.path.exists(java_exec_to_test):
+			java_executable = java_exec_to_test
+
+	p = subprocess.Popen(java_executable + ' -version', shell = True, stdout = subprocess.PIPE, stderr = subprocess.PIPE, universal_newlines=True)
```

Comment:
Why not [`check_output`](https://docs.python.org/3/library/subprocess.html#subprocess.check_output)? Should be easier to use.

Reply:
Yes, much better option. Thanks for the hint.